### ISSUE TITLE
(module name): (short issudeploying helmcharts requires "chart/chart_asset_url" property which isn't taken into account at all when in OCI flowe description)

### ISSUE THREAD

**Issue by psemeniuk** [aws/aws-cdk#22342]
### Describe the bug

Currently when passing OCI url to repository property, the lambda's logic doesn't take into account any of mentioned properties, because chart is pulled from path passed by repository + version properties.

### Expected Behavior

chart/chart_asset_url property isn't required when in OCI flow

### Current Behavior

In OCI flow lambda fails with `chart or chartAsset must be specified` when chart/chart_asset_url property isn't provided.

### Reproduction Steps

Deploy like
```
aws_eks.HelmChart(
            scope=self.scope,
            id="pet-service",
            cluster="some-cluster",
            repository='oci://xxx.dkr.ecr.us-east-1.amazonaws.com/pet-service',
            release='pet-service',
            version='1.0.0'
        )
```

### Possible Solution

Simply put `chart == None and chart_asset_url == None` validation after the OCI-related logic

### CDK CLI Version

2.38.1 (build a5ced21)

### Node.js Version

v16.14.0

### OS

CentOS

### Language

Python

### Language Version

Python 3.8.5

**Comment by pahud** [comment]
@psemeniuk thank you for your report for all the OCI-related issues. I am assigning this to myself and will try reproduce it in my environment. Meanwhile, if you are interested to contribute a PR for this please let me know. You can reach out to me on cdk.dev slack and I will be glad to help you with your pull request.
